### PR TITLE
Added test to assert we are using the specified AZ

### PR DESCRIPTION
**Pull request by fiunchinho**

### CODE REVIEW

File: integration/test/multiaz/multiaz_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+func (s *MultiAZ) Test(ctx context.Context) error {
+	s.logger.LogCtx(ctx, "level", "debug", "message", "getting current availability zones")
+	zones, err := s.provider.GetClusterAZs(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	s.logger.LogCtx(ctx, "level", "debug", "message", "found availability zones", "azs", zones)
+
+	if len(zones) != 1 {
+		return microerror.Mask(wrongAzsUsed)
```
Comment (tuommaki) [inline integration/test/multiaz/multiaz_test.go]: Is this error correct? Should there be something else when AZ's aren't present?

Comment (fiunchinho) [reply integration/test/multiaz/multiaz_test.go]: You mean adding two errors: one when wrong AZ is used, another one when more than one AZs are used?

Comment (tuommaki) [reply integration/test/multiaz/multiaz_test.go]: Yes, or change the error to more generic and use `Maskf(genericError, "problem description")`

Comment (kopiczko) [reply integration/test/multiaz/multiaz_test.go]: Yeah, let's please avoid creating specific errors for everything. The default should be `executionFailedError` + description. Unless you need a matcher. Details here https://github.com/giantswarm/fmt/blob/master/go/errors.md#matching-errors

Comment (fiunchinho) [reply integration/test/multiaz/multiaz_test.go]: ok, I just pushed some changes to fix that.

**Pull request merged**